lib/services/auth_service.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:shared_preferences/shared_preferences.dart';
+
+class AuthService {
+  Future<String> initializedAccount() async {
+    SharedPreferences prefs = await SharedPreferences.getInstance();
```

Comment:
Here each function is asynchronously initializing shared preferences each time it's called.
How acceptable you think this approach is and what are better alternatives we consider to use in this project?
I guess in this particular case it's fine because plugin implemented as singleton, and always returns the same object once it's called first time.
But I am not sure about it, maybe we want to initialize it in main.dart and then pass instance to AuthService?

Reply:
I prefer delaying the initialization until they needed.
and avoiding doing unnecessary initialization in the critical path of the app start that will leed to delaying it starts.
and as you said the SharedPreferences are implemented as a singleton so each invocation is free.
maybe create a top-level function so the invocation is in one line

```
Future<SharedPreferences> prefs() async =>
    await SharedPreferences.getInstance();
```